src/components/calcite-color-hex-input/calcite-color-hex-input.tsx

```diff
@@ -0,0 +1,231 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  h,
+  Method,
+  Prop,
+  State,
+  VNode,
+  Watch
+} from "@stencil/core";
+import {
+  hexChar,
+  hexToRGB,
+  isLonghandHex,
+  isValidHex,
+  normalizeHex,
+  rgbToHex
+} from "../calcite-color/utils";
+import Color from "color";
+import { CSS } from "./resources";
+import { Scale, Theme } from "../../interfaces/common";
+import { RGB } from "../../interfaces/Color";
+import { focusElement, getElementDir } from "../../utils/dom";
+
+const DEFAULT_COLOR = Color();
+
+@Component({
+  tag: "calcite-color-hex-input",
+  styleUrl: "calcite-color-hex-input.scss",
+  shadow: true
+})
+export class CalciteColorHexInput {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+
+  @Element() el: HTMLCalciteColorHexInputElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  //--------------------------------------------------------------------------
+
+  componentWillLoad(): void {
+    const normalized = normalizeHex(this.value);
+
+    if (isValidHex(normalized)) {
+      this.internalColor = Color(normalized);
+      this.value = normalized;
+    }
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Public Properties
+  //
+  //--------------------------------------------------------------------------
+
+  /**
+   * Label used for the hex input.
+   */
+  @Prop()
+  hexLabel = "Hex";
```

Comment:
does this need to be an `intl` named prop?

Reply:
Not sure since the text isn't displayed. For `calcite-icon` we have `textLabel`, which is used specifically for the aria attribute, so these similar props follow that pattern.

Reply:
Since they are strings with default english backup text, I think they probably should follow our `intl-` convention. This is like the modal's close text which is used exclusively for aria-label

Reply:
Alrighty, I'll rename the props and create a `calcite-icon` issue to follow suit. 🎴

Reply:
💥